entrypoint.sh

```diff
@@ -29,27 +29,38 @@ if echo ${INPUT_VERBOSITY} | grep -Eq '[^0-9]'; then
 fi
 echo "Using verbosity level: ${INPUT_VERBOSITY}"
 
-i=0
-while [[ $i -lt $INPUT_VERBOSITY ]]
-do
+for (( i=0; i<${INPUT_VERBOSITY}; i++ )); do
     run_validator="${run_validator} -v"
-    ((i = i + 1))
 done
 
 if [ -n "${INPUT_AS_TYPE}" ]; then
     echo "Validating as type: ${INPUT_AS_TYPE}"
     run_validator="${run_validator} --as-type ${INPUT_AS_TYPE}"
 fi
 
-if [ -n "${INPUT_SKIP_OPTIONAL}" ]; then
-    echo "Skipping optional tests."
-    run_validator="${run_validator} --skip-optional"
-fi
+case ${INPUT_SKIP_OPTIONAL} in
+    y | Y | yes | Yes | YES | true | True | TRUE | on | On | ON)
+        echo "Skipping optional tests."
+        run_validator="${run_validator} --skip-optional"
+        ;;
+    n | N | no | No | NO | false | False | FALSE | off | Off | OFF)
+        ;;
+    *)
+        echo "Non-valid input for 'skip_optional': ${INPUT_SKIP_OPTIONAL}. Will use default (false)."
+        ;;
+esac
 
-if [ -n "${INPUT_FAIL_FAST}" ]; then
-    echo "Will exit on first failure."
-    run_validator="${run_validator} --fail-fast"
-fi
+case ${INPUT_FAIL_FAST} in
+    y | Y | yes | Yes | YES | true | True | TRUE | on | On | ON)
```

Comment:
```suggestion
    y | Y | yes | Yes | YES | true | True | TRUE | on | On | ON | 1)
```

thoughts? Can add 0 for "no" too

Reply:
All these options are the official boolean values for YML files, so I don't think I would use also 1 and 0 unless they are accepted as valid boolean values in YML?

Reply:
https://yaml.org/type/bool.html

Reply:
Fair enough, forgot that the yaml is the main config here and not env vars